libs/eavmlib/src/gpio.erl

```diff
@@ -49,13 +57,27 @@
 ]).
 
 -type gpio() :: pid().
--type pin() :: non_neg_integer() | {atom(), non_neg_integer()}.
--type direction() :: input | output | output_od.
+%% This is the pid returned by `gpio:start/0'.
+-type pin() :: non_neg_integer() | stm32_pin().
+%% The pin definition on the STM32 platform varies from ESP32 and PR2040.
+-type stm32_pin() :: {gpio_bank(), 0..15}.
+%% A pin argument on STM32 is a tuple consisting of a GPIO bank and pin number.
+-type gpio_bank() :: a | b | c | d | e | f | g | h | i | j | k.
+%% STM32 gpio banks vary by board, some only break out `a' thru `h'.
+-type direction() :: input | output | output_od | stm32_direction().
+%% The direction is used to set the mode of operation for a GPIO pin, either as an input, an output, or output with open drain.
+-type stm32_direction() :: {direction(), pull()} | {output, pull(), output_speed()}.
```

Comment:
Similarly here.  Any way to generalize the type name?

Reply:
My reason again for the specic name was that I did not want to mislead users of other platforms into thinking this was a general connivence method that could've used for pin configuration.